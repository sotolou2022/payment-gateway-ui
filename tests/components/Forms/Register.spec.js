import { mount, enableAutoUnmount } from "@vue/test-utils";
import { beforeEach, afterEach, expect } from "@jest/globals";
import { createTestingPinia } from "@pinia/testing";
import Component from "~/components/Forms/Register";

let wrapper;

beforeEach(() => {
  wrapper = mount(Component, { global: { plugins: [createTestingPinia()] } });
});

enableAutoUnmount(afterEach);

describe("Register form test suite", () => {
  it("is a vue compoent", () => expect(wrapper.vm).toBeTruthy());

  it("match snapshot", () => expect(wrapper.html()).toMatchSnapshot());
});
