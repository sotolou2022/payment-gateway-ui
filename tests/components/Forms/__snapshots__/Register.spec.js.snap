// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register form test suite match snapshot 1`] = `
"<form>
  <div class=\\"mb-3 row\\"><label for=\\"name\\" class=\\"col-sm-4 col-form-label text-sm-end\\"> Name: </label>
    <div class=\\"col-sm-8\\"><input type=\\"text\\" class=\\"form-control\\" id=\\"name\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"mb-3 row\\"><label for=\\"email\\" class=\\"col-sm-4 col-form-label text-sm-end\\"> Email: </label>
    <div class=\\"col-sm-8\\"><input type=\\"text\\" class=\\"form-control\\" id=\\"email\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"mb-3 row\\"><label for=\\"inputPassword\\" class=\\"col-sm-4 col-form-label text-sm-end\\"> Password: </label>
    <div class=\\"col-sm-8\\"><input type=\\"password\\" class=\\"form-control\\" id=\\"inputPassword\\" autocomplete=\\"false\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"mb-3 row\\"><label for=\\"inputPassword_confirm\\" class=\\"col-sm-4 col-form-label text-sm-end\\"> Confirm password: </label>
    <div class=\\"col-sm-8\\"><input type=\\"password\\" class=\\"form-control\\" id=\\"inputPassword_confirm\\" autocomplete=\\"false\\"></div>
  </div>
  <div class=\\"mb-3 row\\"><label class=\\"col-sm-4\\"></label>
    <div class=\\"col-sm-8\\"><button type=\\"submit\\" class=\\"btn btn-outline-primary\\"><span>Sign up</span></button></div>
  </div><a to=\\"[object Object]\\" class=\\"text-decoration-none float-end\\"> Already, have an account? </a>
</form>"
`;
